saviSimulation/savi/jason_processing/UxVBehavior.java

```diff
@@ -187,7 +187,7 @@ protected void updatePercepts(PVector mypos) {
 		
 		P.addPerception(new PositionPerception(this.time, (double)myposWithError.x, (double)myposWithError.y, (double)myposWithError.z));
 		//Add velocity
-		P.addPerception(new VelocityPerception(this.time, Math.atan(myposWithError.x/myposWithError.y), 0, this.speedVal));			
+		P.addPerception(new VelocityPerception(this.time, this.compasAngle, 0, this.speedVal));
```

Comment:
This caused a lot of pain and suffering when calculating the absolute/relative positions. Before this change, the bearing did not correlate with the angle that you are facing. I think this is actually supposed to be the intended behaviour but correct me if I'm wrong.

Reply:
Dealt with in another PR #136